lib/ruby_smb/smb2/file.rb

```diff
@@ -115,13 +115,21 @@ def close
       # @raise [RubySMB::Error::InvalidPacket] if the response is not a ReadResponse packet
       # @raise [RubySMB::Error::UnexpectedStatusCode] if the response NTStatus is not STATUS_SUCCESS
       def read(bytes: size, offset: 0)
-        atomic_read_size = if bytes > tree.client.server_max_read_size
-                             tree.client.server_max_read_size
+        max_read = tree.client.server_max_read_size
+        if tree.client.dialect == '0x0202' || !tree.client.server_supports_multi_credit
+          max_read = 65536
+        end
+        atomic_read_size = if bytes > max_read
+                             max_read
                            else
                              bytes
                            end
```

Comment:
I think using [Enumerable#min](https://ruby-doc.org/core-2.7.1/Enumerable.html#method-i-min) here would be a bit cleaner.

```suggestion
        atomic_write_size = [bytes, max_write].min
```